From 7373850a34c3b0da185312a04fe28a12e48e645a Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Tue, 10 Mar 2020 19:59:04 +0200
Subject: [PATCH 19/26] Add screenshot sounds settings

  * from
  https://github.com/Havoc-OS/android_packages_apps_Settings/commit/52730f137e12910589cb3074bd2c6587e5bbb96a

	modified:   res/values/strings.xml
	modified:   res/xml/sound_settings.xml
	new file:   src/com/android/settings/notification/ScreenshotSoundPreferenceController.java
	modified:   src/com/android/settings/notification/SoundSettings.java
---
 res/values/strings.xml                        |  3 ++
 res/xml/sound_settings.xml                    |  5 +++
 .../ScreenshotSoundPreferenceController.java  | 37 +++++++++++++++++++
 .../settings/notification/SoundSettings.java  |  5 ++-
 4 files changed, 49 insertions(+), 1 deletion(-)
 create mode 100644 src/com/android/settings/notification/ScreenshotSoundPreferenceController.java

diff --git a/res/values/strings.xml b/res/values/strings.xml
index bce54092b7..9de472e20b 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -11458,4 +11458,7 @@
     <string name="back_options_dialog_title">Options</string>
     <string name="back_height_message">Amount of screen height used as touchable region for back gesture</string>
 
+    <!-- Screenshot sound -->
+    <string name="screenshot_sound_title">Screenshot sound</string>
+
 </resources>
diff --git a/res/xml/sound_settings.xml b/res/xml/sound_settings.xml
index d99cc00c8d..23e4fe3a3d 100644
--- a/res/xml/sound_settings.xml
+++ b/res/xml/sound_settings.xml
@@ -152,6 +152,11 @@
           android:key="dial_pad_tones"
           android:title="@string/dial_pad_tones_title"/>
 
+        <!-- Screenshot sounds -->
+        <SwitchPreference
+          android:key="screenshot_sounds"
+          android:title="@string/screenshot_sound_title" />
+
         <!-- Screen locking sounds -->
         <SwitchPreference
           android:key="screen_locking_sounds"
diff --git a/src/com/android/settings/notification/ScreenshotSoundPreferenceController.java b/src/com/android/settings/notification/ScreenshotSoundPreferenceController.java
new file mode 100644
index 0000000000..18ce39fc78
--- /dev/null
+++ b/src/com/android/settings/notification/ScreenshotSoundPreferenceController.java
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settings.notification;
+
+import static com.android.settings.notification.SettingPref.TYPE_SYSTEM;
+
+import android.content.Context;
+
+import android.provider.Settings.System;
+import com.android.settings.SettingsPreferenceFragment;
+import com.android.settingslib.core.lifecycle.Lifecycle;
+
+public class ScreenshotSoundPreferenceController extends SettingPrefController {
+
+    private static final String KEY_SCREENSHOT_SOUNDS = "screenshot_sounds";
+
+    public ScreenshotSoundPreferenceController(Context context, SettingsPreferenceFragment parent,
+            Lifecycle lifecycle) {
+        super(context, parent, lifecycle);
+        mPreference = new SettingPref(
+            TYPE_SYSTEM, KEY_SCREENSHOT_SOUNDS, System.SCREENSHOT_SOUND, DEFAULT_ON);
+    }
+}
diff --git a/src/com/android/settings/notification/SoundSettings.java b/src/com/android/settings/notification/SoundSettings.java
index eec0fb8018..5857aa7297 100644
--- a/src/com/android/settings/notification/SoundSettings.java
+++ b/src/com/android/settings/notification/SoundSettings.java
@@ -258,6 +258,8 @@ public class SoundSettings extends DashboardFragment implements OnActivityResult
                 new BootSoundPreferenceController(context);
         final EmergencyTonePreferenceController emergencyTonePreferenceController =
                 new EmergencyTonePreferenceController(context, fragment, lifecycle);
+        final ScreenshotSoundPreferenceController screenshotSoundPreferenceController =
+                new ScreenshotSoundPreferenceController(context, fragment, lifecycle);
 
         controllers.add(dialPadTonePreferenceController);
         controllers.add(screenLockSoundPreferenceController);
@@ -268,6 +270,7 @@ public class SoundSettings extends DashboardFragment implements OnActivityResult
         controllers.add(dockAudioMediaPreferenceController);
         controllers.add(bootSoundPreferenceController);
         controllers.add(emergencyTonePreferenceController);
+        controllers.add(screenshotSoundPreferenceController);
         controllers.add(new PreferenceCategoryController(context,
                 "other_sounds_and_vibrations_category").setChildren(
                 Arrays.asList(dialPadTonePreferenceController,
@@ -318,4 +321,4 @@ public class SoundSettings extends DashboardFragment implements OnActivityResult
             mDialogFragment.onListPreferenceUpdated(preference);
         }
     }
-}
\ No newline at end of file
+}
-- 
2.17.1

